recipes/newrelic/infrastructure/super-agent/debian.yml

```diff
@@ -519,58 +519,143 @@ install:
-              fi
+                  "$TOKEN_RENEWAL_ENDPOINT"
+                )
 
-              echo "Error creating the new system identity. The API endpoint returned $HTTP_CODE. Retrying ($RETRY/3)..."
-              sleep 2
-            done
+                if [ $HTTP_CODE -eq 200 ]; then
+                  break
+                fi
+          
+                ERROR_MESSAGE=$(/usr/local/bin/newrelic utils jq '.error_description // "invalid_request"' < "$TEMPORAL_FOLDER/response_token.json" | tr -d '"')
 
-            if [ $HTTP_CODE -ne 200 ]; then
-              exit 99
-            fi
+                echo "Error getting system identity auth token. The API endpoint returned $HTTP_CODE: $ERROR_MESSAGE. Retrying ($RETRY/3)..."
+                sleep 2
+              done
 
-            ERROR_MESSAGE=$(/usr/local/bin/newrelic utils jq '.errors[0].message // "NOERROR"' < "$TEMPORAL_FOLDER/response.json" | tr -d '"')
-            if [ "$ERROR_MESSAGE" != "NOERROR" ]; then
-              echo "Failed to create a New Relic System Identity for Fleet Control communication authentication. Please verify that your User Key is valid and that your Account Organization has the necessary permissions to create a System Identity: $ERROR_MESSAGE"
-              exit 100
+              if [ $HTTP_CODE -ne 200 ]; then
+                echo "Error getting system identity auth token"
+                exit 99
+              fi
+              
+              ACCESS_TOKEN=$(/usr/local/bin/newrelic utils jq  '.access_token' < "$RESPONSE_FILE" | tr -d '"' )
+          
+              ############################################################
+              # Create System Identity
+              ############################################################
+              DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+              NAME="System Identity for $(hostname) - $DATE"
+  
+              for RETRY in 1 2 3; do
+                HTTP_CODE=$(echo '{ "query":
+                    "mutation {
+                      create(
+                        name: \"'$NAME'\",
+                        organizationId: \"{{ .NEW_RELIC_ORGANIZATION }}\",
+                        publicKey: \"'$(openssl enc -base64 -A -in "$TEMPORAL_FOLDER/pub")'\"
+                      ) {
+                        clientId,
+                        name
+                      }
+                    }"
+                  }' | tr -d $'\n' | curl \
+                    -s -w "%{http_code}" \
+                    -H "Content-Type: application/json" \
+                    -H "Authorization: Bearer $ACCESS_TOKEN" \
+                    -o "$TEMPORAL_FOLDER/response.json" \
+                    --data-binary @- \
+                    "$IDENTITY_CREATION_ENDPOINT"
+                )
+  
+                if [ $HTTP_CODE -eq 200 ]; then
+                  break
+                fi
+  
+                echo "Error creating the new system identity. The API endpoint returned $HTTP_CODE. Retrying ($RETRY/3)..."
```

Comment:
Would it make sense to use slightly different error messages in order to help troubleshooting during the transition? For instance we could add L2 `Error creating L2 system identity...`

Reply:
Good point! Changed in f6777411fa6b142a97ef758cb7ba86e18bf63739